keps/sig-cluster-lifecycle/kubeadm/20191229-kubeadm-move-out-of-kk.md

```diff
@@ -0,0 +1,594 @@
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed](#infrastructure-needed)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone,
+which links to KEP (this should be a link to the KEP location
+in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and
+SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website],
+for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents,
+links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+This document outlines the proposal for moving kubeadm outside of the
+kubernetes/kubernetes repository into the kubernetes/kubeadm
+repository.
+
+A proposal was already done in the following Google Doc and it
+received a healthy discussion:
+https://docs.google.com/document/d/1xFAGAjfPQZrcqmapTwpJR5aUA7-2ku1asVeq3dBrjnQ
+
+## Motivation
+
+The monolith of kubernetes/kubernetes (k/k) is splitting and components
+such as kubeadm should reside in designated repositories. Multiple
+discussions during 2019 happened where the SIG Architecture Code
+Organization sub project suggested that it is time for kubeadm to
+move and that the kubeadm maintainers should start investigating
+what has to be done for this to happen.
+
+Moving kubeadm will benefit both kubernetes/kubernetes and the kubeadm
+maintainers:
+- Less noise in k/k from kubeadm issues and PRs.
+- Make it easier for the kubeadm maintainers to track kubeadm PRs.
+- No need to run the full suite of k8s pre-submit e2e jobs on kubeadm PRs.
+
+### Goals
+
+- Move the kubeadm source code to the k/kubeadm repository.
+- Preserve the current kubeadm versioning, matching the k8s versioning.
+- Continue packing kubeadm as part of the "node" tarball for Windows
+and Linux.
+- Continue having kubeadm test jobs as release-informing (-blocking in the future).
+- Continue testing kubeadm using the existing set of tools - e.g. kinder.
+- Continue to release kubeadm documentation as part of the k8s.io website
+every k8s release.
+
+### Non-Goals
+
+- Decouple kubeadm from the k8s versioning. This can only create confusion
+for the users and those depending on k8s/kubeadm version skew.
+- Not releasing kubeadm artifacts for end users to consume.
+- Use Bazel for building kubeadm once in the k/kubeadm repository.
+Multiple maintainers have expressed a negative vote on Bazel.
+- Refactor the kubeadm source structure to easily distinguish between
+frontend and backend. This work is out of scope for this proposal.
```

Comment:
Does this non-goal also mean that using something like `internal` for the new `kubeadm` structure is out of scope for this proposal?

Reply:
yes. that is a separate KEP / refactor.